src/Analyzer.php

```diff
@@ -81,6 +86,28 @@ public function registerStatementPasses(array $statementPasses)
     }
 
     /**
+     * @param array $scalarPasses all the scalar analyzers
+     * @throws \RuntimeException if the analyzer does not implement the required interface
+     */
+    public function registerScalarPasses(array $scalarPasses)
+    {
+        foreach ($scalarPasses as $pass) {
+            if (!$pass instanceof AnalyzerPassInterface) {
+                throw new \RuntimeException('Analyzer pass must implement AnalyzerPassInterface');
+            }
+
+            $bindOnScalars = $pass->getRegister();
+            foreach ($bindOnScalars as $bindOnScalar) {
+                if (isset($this->bindOnScalars[$bindOnScalar])) {
+                    $this->bindOnScalars[$bindOnScalar][] = $pass;
+                } else {
+                    $this->bindOnScalars[$bindOnScalar] = [$pass];
+                }
+            }
+        }
+    }
+
+    /**
      * binds the listeners
      */
     public function bind()
```

Comment:
you need to bind your listener here. If you do this correctly it works (I checked it)